lib/engine/game/g_18_rhl/game.rb

```diff
@@ -623,79 +641,167 @@ def event_remove_tile_block!
           yellow_block_hex.tile.icons.reject! { |i| i.name == 'green_hex' }
         end
 
+        def illegal_revisit_of_rhine_metropolis_hex?(actual_visits)
+          actual_visits.select { |v| RHINE_METROPOLIS_HEXES.include?(v.hex.name) && v.hex.tile.color == :green }
+                       .group_by { |v| v.hex.name }.reject { |_, e| e.size < 2 }
+                       .any?
```

Comment:
it should be group_by.any? {...} not reject? any

Reply:
Fixed